### PR TITLE
fix(eventstream): Handle SIGTERM in post-processing forwarder

### PR DESCRIPTION
**Pull request by tkaemming**
This change makes the post-processing forwarder handle `SIGTERM` to initiate a graceful shutdown in addition to `SIGINT`. This ensures a greater variety of process managers (i.e. Kubernetes) can safely shutdown the consumer processes, resulting in a faster rebalancing procedure when processes are stopped under normal operating circumstances: for example, during deploys or consumer group downscaling.

See also: getsentry/snuba#396

### CODE REVIEW

**Review by JTCunning — COMMENTED** [review]
Your call on this...
While we're in here, might make sense to upgrade the partition assignment/revocation/shutdown logs to `info` if we want to do future introspection.

**Review by dcramer — APPROVED** [review]
first time ive seen nonlocal in python.. guessing it just finds first ref in outer scopes?

File: src/sentry/eventstream/kafka/backend.py [hunk 1]
```diff
@@ -181,38 +183,45 @@ def commit_offsets():
...
-                if message is None:
-                    continue
+        shutdown_requested = False
 
-                error = message.error()
-                if error is not None:
-                    raise Exception(error)
+        def handle_shutdown_request(signum: int, frame: Any) -> None:
+            nonlocal shutdown_requested
+            logger.debug("Received signal %r, requesting shutdown...", signum)
```
Comment (dcramer) [inline src/sentry/eventstream/kafka/backend.py:190]: only feedback id have is if this happens irregularly, maybe make this _at least_ info

**Review by fpacifici — APPROVED** [review]
Hi, ted.

This is good to me.
I am curious on how were we used to stop this consumer before ?

**Comment by alxu-sentry** [comment]
can't comment on the logic here, but using `signal.signal` in this way seems a little un-Pythonic. maybe we could use something more like https://stackoverflow.com/a/24574672 with a try/finally block? that would also handle properly shutting down in case of other exceptions.

File: src/sentry/eventstream/kafka/backend.py [hunk 2]
```diff
@@ -106,7 +108,7 @@ def commit(partitions):
             return results
 
         def on_assign(consumer, partitions):
-            logger.debug("Received partition assignment: %r", partitions)
+            logger.info("Received partition assignment: %r", partitions)
```
[the remark this answers is not in the document]
Comment (JTCunning) [reply src/sentry/eventstream/kafka/backend.py:111]: https://github.com/getsentry/sentry/pull/24244#pullrequestreview-604466650

**Comment by JTCunning** [comment]
@fpacifici: We stopped this consumer before with the signal it successfully caught. It is now a different signal in Kubernetes.

**Pull request merged**